rcl/src/rcl/wait.c

```diff
@@ -79,6 +80,9 @@ __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
     assert(ret == RCL_RET_OK);  // Defensive, shouldn't fail with size 0.
   }
   if (wait_set->impl) {
+    if (wait_set->impl->rmw_waitset) {
+      assert(rmw_destroy_waitset(wait_set->impl->rmw_waitset) == RMW_RET_OK);
```

Comment:
I agree, I think checking it in debug only is ok (that was the idea with the assert anyways), but then it results in a unused variable warnings which requires us to but in that obnoxious other "no lint" line. I'm not sure whether or not the performance difference is appreciable.